### PR TITLE
Support Client Filters Pass in the JDK client

### PR DESCRIPTION
**Pull request by timyates**
Enables Http Client Filters for the JDK client.

### CODE REVIEW

**Comment by timyates** [comment]
Fixed up the logging as per https://github.com/micronaut-projects/micronaut-core/pull/9710 (so we don't lose it in the merge)

File: http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java [hunk 1]
```diff
@@ -312,4 +343,64 @@ private <I> Mono<URI> resolveURI(io.micronaut.http.HttpRequest<I> request) {
...
+        io.micronaut.http.HttpRequest<?> request,
+        Argument<O> bodyType
+    ) {
+        return Flux.defer(() -> mapToHttpRequest(request, bodyType)) // defered so any client filter changes are used
+            .map(httpRequest -> {
+                if (log.isDebugEnabled()) {
+                    log.debug("Client {} Sending HTTP Request: {}", clientId, httpRequest);
+                }
+                if (log.isTraceEnabled()) {
+                    httpRequest.headers().map().forEach((k, v) -> log.trace("Client {} Sending HTTP Request Header: {}={}", clientId, k, v));
```
Comment (sdelamo) [inline http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java]: let's use `HttpHeaderUtils` to take into account logging `HEADER_MASK_PATTERNS`

Comment (sdelamo) [reply http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java]: I created a PR https://github.com/micronaut-projects/micronaut-core/pull/9715

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by yawkat — APPROVED** [review]

File: http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java (hunk 1 above)
Comment (timyates) [reply http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java]: 👍

**Review by sdelamo — APPROVED** [review]

**Pull request merged**